client/sites/components/sites-dataviews/index.tsx

```diff
@@ -73,13 +80,27 @@ const DotcomSitesDataViews = ( {
 			if ( selectedSiteIds.length === 0 ) {
 				return;
 			}
+
 			const site = sites.find( ( s ) => s.ID === Number( selectedSiteIds[ 0 ] ) );
 			if ( site && ! site.is_deleted ) {
-				sitePreviewPane.open( site, 'list_row_click' );
+				const isAdmin = canCurrentUser( state, site.ID, 'manage_options' );
```

Comment:
What about extracting this logic and the duplicated one into an util function? Otherwise I'm afraid it will happen again in the future

https://github.com/Automattic/wp-calypso/blob/670042ccfb2a7681102b9c571b84938c1d5e0a86/client/sites/components/sites-dataviews/site-icon.tsx#L38-L46

Reply:
Oh true, we have the same logic in 3-4 places 😅